server/models/postgis/task.py

```diff
@@ -179,6 +194,29 @@ def get_tasks(project_id: int, task_ids: List[int]):
         """ Get all tasks that match supplied list """
         return Task.query.filter(Task.project_id == project_id, Task.id.in_(task_ids))
 
+    @staticmethod
+    def auto_unlock_tasks(project_id: int):
+        """Unlock all tasks locked more than 2 hours ago"""
+        old_locks_query = '''select t.id
+             from tasks t,
+                  task_history th
+             where t.project_id = {0}
+             and t.id = th.task_id
+             and t.project_id = th.project_id
+             and t.task_status in (1,3)
+             and th.action in ( 'LOCKED_FOR_VALIDATION','LOCKED_FOR_MAPPING' )
+             and th.action_text is null
+             and th.action_date <= current_timestamp - INTERVAL '2 hours' '''.format(project_id)
+
+        old_tasks = db.engine.execute(old_locks_query)
+
+        if old_tasks.rowcount == 0:
+            return
```

Comment:
Add a comment here explaining that no tasks older than 2 hours found so return without further processing